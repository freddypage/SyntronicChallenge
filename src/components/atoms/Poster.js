import React, {useState, Component} from 'react';
import {Image, Text, TextInput, View, ActivityIndicator} from 'react-native';
import { styles, buttons } from 'styles/style';


class Poster extends Component {

    componentDidMount() {
        this.setState({isLoading:false});
        return
    }

    constructor(props){
        super(props);
        this.state = {
            isLoading: true
        }
    }

    render() {
        const {isLoading} = this.state;
        return(
            <View style={styles.centered}>
                {isLoading ? <ActivityIndicator/> : 
            (<Image resizeMode={'cover'} style={styles.image} 
                source={{uri:"https://image.tmdb.org/t/p/original"+this.props.url}}
                />
            )}
            </View>
        );
    }
}

export default Poster;