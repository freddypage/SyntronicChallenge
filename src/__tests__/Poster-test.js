import React from 'react';
import Poster from 'atoms/Poster';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<Poster />).toJSON();
  expect(tree).toMatchSnapshot();
});