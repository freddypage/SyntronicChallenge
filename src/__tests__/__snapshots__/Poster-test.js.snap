// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "textAlign": "center",
    }
  }
>
  <Image
    resizeMode="cover"
    source={
      Object {
        "uri": "https://image.tmdb.org/t/p/originalundefined",
      }
    }
    style={
      Object {
        "borderRadius": 20,
        "height": 200,
        "width": 135,
      }
    }
  />
</View>
`;
